### ISSUE TITLE
Issue with intersection of circle and box

### ISSUE THREAD

**Issue by Saturn49** [alexbol99/flatten-boolean-op#2]
I'm having some trouble using intersection, and boiled down one of my issues to a simple test case.  I construct a Polygon as a circle (arc from 0 to 2*PI) and a box (4 points) and intersect them.  The results seem nonsensical to me.  The first two segments are (50,950) and (950,50), completely back-tracking.

Am I doing something wrong?  

Here's a simple repro:
```
var booleanOp = require("flatten-boolean-op" )
var flattenJs = require("flatten-js")

let { arc, circle, line, matrix, point, ray, ray_shoot, segment, vector, Polygon } = flattenJs;
let { unify, intersect, subtract } = booleanOp;

let myBox = new flattenJs.Polygon();
myBox.addFace([point(50, 50), point(50,950), point(950, 950), point(950, 50)]); 

let myCircle = new flattenJs.Polygon();
myCircle.addFace([arc(point(0,0), 1100, 0, 2*Math.PI)]);

let myIntersection = intersect(myBox, myCircle);

console.log(myIntersection);
```

**Comment by alexbol99** [comment]
Hi @Saturn49 ,

You are doing all right, and the result is correct:

![intersect_before](https://user-images.githubusercontent.com/6965440/45995402-a88e5280-c0a0-11e8-8525-de77eb86aa3d.JPG)
![intersect_after](https://user-images.githubusercontent.com/6965440/45995400-a62bf880-c0a0-11e8-9e2e-30ba75e7bd27.JPG)

Please see live example in this [notebook](https://beta.observablehq.com/@alexbol99/boolean-operations-test-issue-2)

**Comment by Saturn49** [comment]
Thanks Alex, you're right that one is working correctly - I had a bug in my drawing code and I got confused when debugging from the raw segments.  Thank you for making this library, it has already saved me a ton of time!

However, I did manage to reproduce an issue on Observable, this one is more complete with what I'm trying to do:

https://beta.observablehq.com/@saturn49/boolean-operations-test-issue-2

The basic operation is intersecting and subtracting circles from a starting box.  The last subtraction (A-B) doesn't seem to be working correctly and ends up with an empty Polygon:

https://www.screencast.com/t/AOhhqQizt

--Nathan

**Comment by Saturn49** [comment]
Thanks Alex!  I was able to test with the new version and I can confirm things are working as expected now.  It isn't a problem to keep all polygons clockwise in my application.  Also, thanks for pointing out the Y-value / clockwise issue.  I'm drawing directly onto an html Canvas element which is oriented Y-axis from top to bottom, so that explains why I had to invert the counterClockwise field from the polygon when drawing the arcs on the Canvas.
